src/common/ZulipTextButton.js

```diff
@@ -0,0 +1,109 @@
+/* @flow strict-local */
+import React, { useMemo } from 'react';
+import { View } from 'react-native';
+
+import type { LocalizableText } from '../types';
+import { BRAND_COLOR, createStyleSheet } from '../styles';
+import { Label } from '.';
+import Touchable from './Touchable';
+
+// TODO(?): Support, e.g., "small", "medium", "large" if it makes
+// sense. It probably doesn't.
+type Variant = 'standard';
```

Comment:
What would small, medium, or large mean? I don't see those at https://material.io/components/buttons .

Reply:
I don't have particular variants in mind right now, but I wanted to pretty firmly give a good path of least resistance in case we come up with some. In particular, a path that would have us identify a legitimate variant and write its style in a coherent way. A path that doesn't look like "hmm let's add a flag to allow callers to specify leopard print, and another for this one caller where it's running a little too close to a FAB so it needs to scrunch up its character spacing a bit", etc.

Does this approach to that seem useful?

Reply:
I see -- so IIUC the idea is to try to avoid this situation that `ZulipButton` has:
```js
  style?: SubsetProperties<
    ViewStyle,
    {|
      // The use of specific style properties for callers to micromanage the
      // layout of the button is pretty chaotic and fragile, introducing
      // implicit dependencies on details of the ZulipButton implementation.
      // TODO: Assimilate those usages into a more coherent set of options
      //   provided by the ZulipButton interface explicitly.
      //
      // Until then, by listing here the properties we do use, we at least
      // make it possible when working on the ZulipButton implementation to
      // know the scope of different ways that callers can mess with the
      // styles.  […]
      /* … a list of 14 particular style props … */
```

That is a very sensible thing to want to avoid!

I guess maybe just replace the comment with an explanation more like the one you just gave? Instead of these examples that the comment acknowledges probably don't make sense :slightly_smiling_face: